controllers/managedocs_controller.go

```diff
@@ -1669,6 +1681,13 @@ func (r *ManagedOCSReconciler) findOCSVolumeClaims() (bool, error) {
 	return false, nil
 }
 
+func (r *ManagedOCSReconciler) hasOCSStorageConsumers() (bool, error) {
+	storageConsumerList := ocsv1alpha1.StorageConsumerList{}
```

Comment:
- it seems `kind: StorageConsumer` isn't part of either `Watches`/`Owns`and I could see below issues with that:
1. straightforward, we don't get a reconcile event when resource changes
2. no cache is built for this object in operator-sdk

- if we want ocs-operator to be the sole owner I'd suggest using `r.UnrestrictedClient`
- just confirming, `storageconsumer` will always be created in deployer namespace right?

minor optimisation: since we want presence or absence why not restrict the list to only one item?
```
r.Client.List(ctx, storageConsumerList, &client.ListOptions{Limit: int64(1)})
```

Reply:
> * it seems `kind: StorageConsumer` isn't part of either `Watches`/`Owns`and I could see below issues with that:
> 
> 1. straightforward, we don't get a reconcile event when resource changes
Do we even need to watch this, since this should only reconcile during uninstallation
> 2. no cache is built for this object in operator-sdk
I don't think we need a cache for this. Maybe @nb-ohad can clarify
> 
> * if we want ocs-operator to be the sole owner I'd suggest using `r.UnrestrictedClient`
Why? It will be in the same namespace
> * just confirming, `storageconsumer` will always be created in deployer namespace right?
Yes
> 
> minor optimisation: since we want presence or absence why not restrict the list to only one item?
> 
> ```
> r.Client.List(ctx, storageConsumerList, &client.ListOptions{Limit: int64(1)})
> ```
Yeah! this change seems more optimised. Thanks.

Reply:
@leelavg 
+1 for the suggestion to limit the list results to 1. 
I think that the `hasOCSVolumes` util can benefit from this enhancement as well.

Regarding the `watch` or using `unrestrictedClient` I don't really understand why should we do it. 
This code will only be invoked when we are instructed to uninstall the offering and act as a guard, in the case of condition failure we are requeuing with a very specific "after" time (10 seconds) to try again. 
This means no watch is needed as there is no point in trying to count and store a counter of creations and deletions of storage consumers throughout the lifetime of the deployment for a single check when we want to uninstall. 

To clarify, a watch is usually used in two very specific use cases:
1. Watch input resources to get changes in the desired state
2. Watch reconciled resources (output resources) to revert changes to them if they are changed by another agent and divert from the desired state

Out use case here does not match any of the two

Reply:
> Out use case here does not match any of the two
- ack, my understanding might be wrong here then
- for posterity below is what I was thinking 

1. When we watch/own resources, sdk creates cache for us and thought maintaining that as well will be useful even during uninstall 
2. If there's a cache miss there might be a cool off period after which the request will be sent to api-server but not immediately

- from my comment pls implement only limiting results and pls discard other requests based on @nb-ohad reasoning

Reply:
@leelavg 
I think you are mixing between the k8s client cache (for clients that were created with the cache option), which to the best of my understanding, has nothing to do with controller watch. 
And the way that op-sdk decides when to implement controller watch, which internally uses the k8s client.

The content of the client cache is based on the client usage (mainly `get`, and `list` operations, but also `watch`), but might also have some relation to the controller watch which is implemented using the client semantics. 

In any case, I still think my reasoning above (last comment) stands still.

Reply:
Done